pkg/apis/build/v1alpha1/build_validation.go

```diff
@@ -31,16 +31,21 @@ func (b *Build) Validate() *apis.FieldError {
 // Validate for build spec
 func (bs *BuildSpec) Validate() *apis.FieldError {
 	if bs.Template == nil && len(bs.Steps) == 0 {
-		return apis.ErrMissingField("b.spec.template").Also(apis.ErrMissingField("b.spec.steps"))
+		return apis.ErrMissingOneOf("template", "steps")
 	}
 	if bs.Template != nil && len(bs.Steps) > 0 {
-		return apis.ErrMissingField("b.spec.template").Also(apis.ErrMissingField("b.spec.steps"))
+		return apis.ErrMultipleOneOf("template", "steps")
 	}
-
+	//removing below conditional because bs.Template.Name is already being checked in bs.Template.Validate()
+	/*
 	if bs.Template != nil && bs.Template.Name == "" {
-		apis.ErrMissingField("build.spec.template.name")
-	}
+		apis.ErrMissingField("name").ViaField("template")
+	}*/
 
+
+	/*below method potentially has a bug:
+		it does not Validate if only a "Source" has been set, it only validates if multiple sources have been set
+	*/
 	if err := bs.validateSources(); err != nil {
```

Comment:
For many of the following checks, we can actually group them all up with `.Also()`, which is designed to handle nil, so you don't need a giant chain of `nil` checks.

Reply:
Again, thanks for the heads up; That definitely sounds like a cleaner way to implement this part!